lib/cf-socket.c

```diff
@@ -244,6 +244,13 @@ void Curl_sock_assign_addr(struct Curl_sockaddr_ex *dest,
     dest->socktype = SOCK_STREAM;
     dest->protocol = IPPROTO_IP;
     break;
+  case TRNSPRT_MPTCP:
+#ifndef IPPROTO_MPTCP
```

Comment:
Wouldn't it be more clean to rather `result = CURLE_UNSUPPORTED_PROTOCOL;` if the define is missing?
If the idea is to hide the error quietly, then do so, rather than using hardcoded value (which may or may not be the same on all platforms).

Reply:
> If we want an explicit error to say that the feature is not supported, maybe it would be better to do that in an `#else` after the `#ifdef __linux__` we added in `Curl_cf_tcp_create()`, no?

Something like this maybe?

```diff
diff --git a/lib/cf-socket.c b/lib/cf-socket.c
index a852a5231..8bc5423d9 100644
--- a/lib/cf-socket.c
+++ b/lib/cf-socket.c
@@ -1610,10 +1610,14 @@ CURLcode Curl_cf_tcp_create(struct Curl_cfilter **pcf,
     goto out;
   }
 
+  if(data->set.mptcp) {
 #ifdef __linux__
-  if(data->set.mptcp)
     transport = TRNSPRT_MPTCP;
+#else
+    result = CURLE_UNSUPPORTED_PROTOCOL;
+    goto out;
 #endif
+  }
 
   cf_socket_ctx_init(ctx, ai, transport);
 

```

Reply:
FYI, I just pushed a new version have this diff (+ a rebase on top of master)